perses/dispersed/feptasks.py

```diff
@@ -534,148 +404,242 @@ def run_protocol(equilibrium_result: EquilibriumResult, thermodynamic_state: sta
-        cw_filepath_parts[-1] = "cw.npy"
-        pw_filepath_parts[-1] = "pw.npy"
-
-        cum_work_filepath = ".".join(cw_filepath_parts)
-        prot_work_filepath = ".".join(pw_filepath_parts)
-
         #if writing configurations was requested, get the trajectory
         if write_configuration:
-            try:
-                trajectory = ne_mc_move.trajectory
-                write_nonequilibrium_trajectory(nonequilibrium_result, trajectory, trajectory_filename)
-            except NoTrajectoryException:
-                pass
-
-        np.save(cum_work_filepath, nonequilibrium_result.cumulative_work)
-        np.save(prot_work_filepath, nonequilibrium_result.protocol_work)
+                trajectory = ne_mc_move._trajectory
+                write_nonequilibrium_trajectory(trajectory, trajectory_filename)
+                _logger.debug(f"successfully wrote nonequilibrium trajectory to {trajectory_filename}")
 
-        if measure_shadow_work:
-            np.save(shad_work_filepath, shadow_work)
+    if not timer:
+        timers = {}
+    else:
+        timers = ne_mc_move._timers
+
+    if ne_mc_move.succeed:
+        neq_result = NonequilibriumResult(succeed = True,
+                                          sampler_state = sampler_state,
+                                          protocol_work = protocol_work,
+                                          cumulative_work = cumulative_work,
+                                          shadow_work = shadow_work,
+                                          kinetic_energies = kinetic_energies,
+                                          timers = timers)
+    else:
+        neq_result = NonequilibriumResult(succeed = False,
+                                          sampler_state = sampler_state,
+                                          protocol_work = protocol_work,
+                                          cumulative_work = cumulative_work,
+                                          shadow_work = shadow_work,
+                                          kinetic_energies = kinetic_energies,
+                                          timers = timers)
 
-    return nonequilibrium_result
+    return neq_result
 
-def run_equilibrium(equilibrium_result: EquilibriumResult, thermodynamic_state: states.ThermodynamicState,
+def run_equilibrium(thermodynamic_state: states.CompoundThermodynamicState, eq_result: EquilibriumResult,
                     nsteps_equil: int, topology: md.Topology, n_iterations : int,
-                    atom_indices_to_save: List[int] = None, trajectory_filename: str = None, splitting: str="V R O R V", timestep: unit.Quantity=1.0*unit.femtoseconds) -> EquilibriumResult:
+                    atom_indices_to_save: List[int] = None, trajectory_filename: str = None,
+                    splitting: str="V R O R V", timestep: unit.Quantity=1.0*unit.femtoseconds,
+                    max_size: float=1024*1e3, timer: bool=False, _minimize: bool = False, file_iterator: int = 0,
```

Comment:
I think it's a little bit unusual to pass a variable into a function as private... is there a reason that it's private?